Scoop/Controller/Profile/EditProfileViewController.swift

```diff
@@ -0,0 +1,294 @@
+//
+//  EditProfileViewController.swift
+//  Scoop
+//
+//  Created by Caitlyn Jin on 9/27/23.
+//
+
+import SDWebImage
+import UIKit
+
+class EditProfileViewController: UIViewController {
+    
+    // MARK: - Views
+    
+    private let classTextField = UITextField()
+    private let hometownTextField = UITextField()
+    private let nameTextField = UITextField()
+    private let profileImageView = UIImageView()
+    private let pronounsTextField = UITextField()
+    private let stackView = UIStackView()
+    private let uploadPhotoButton = UIButton()
+    
+    weak var delegate: ProfileViewDelegate?
+    
+    // MARK: - User Data
+    
+    private var user: BaseUser?
+    
+    private var hometown: String?
+    private var music: Float?
+    private var snack: String?
+    private var song: String?
+    private var stop: String?
+    private var talkative: Float?
+    
+    // MARK: - Lifecycle Functions
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.backgroundColor = UIColor.white
+        
+        setupHeader()
+        setupProfileImageView()
+        setupUploadPhotoButton()
+        setupStackView()
+    }
+    
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        
+        profileImageView.layer.cornerRadius = profileImageView.frame.size.width / 2
+        uploadPhotoButton.layer.cornerRadius = uploadPhotoButton.frame.size.width / 2
+    }
+    
+    // TODO: Temporary while save button is not implemented
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        
+        getUserPreferences()
+        updateUser()
+    }
+    
+    // MARK: - Initializers
+    
+    init(user: BaseUser, hometown: String) {
+        super.init(nibName: nil, bundle: nil)
+        self.user = user
+        
+        if let profilePicUrl = user.profilePicUrl {
+            profileImageView.sd_setImage(with: URL(string: profilePicUrl))
+        } else {
+            profileImageView.image = UIImage.emptyImage
+        }
+        
+        nameTextField.text = "\(user.firstName) \(user.lastName)"
+        pronounsTextField.text = user.pronouns
+        hometownTextField.text = hometown
+        classTextField.text = user.grade
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    // MARK: - Setup View Functions
+    
+    private func setupHeader() {
+        let dottedLineMultiplier = 0.52
+        let solidLineVerticalInset = -12.75
+        let solidLineMultiplier = 0.32
+        let screenSize = UIScreen.main.bounds
+        let dottedline = UIImageView(image: UIImage.dottedLine)
+        let solidline = UIView()
+        
+        dottedline.contentMode = .scaleAspectFit
+        view.addSubview(dottedline)
+        
+        dottedline.snp.makeConstraints { make in
+            make.height.equalTo(1)
```

Comment:
Just curious, is it necessary to make the height equal to 1? I feel like that makes it barely visible.

Reply:
Pretty sure that's just how the design looks, it's the same on other views